pkg/npm/package_manager.go

```diff
@@ -87,3 +87,58 @@ func (exec *Execute) detectPackageManager() (*PackageManager, error) {
 		InstallArgs:    []string{"install"},
 	}, nil
 }
+
+// setupPnpm handles the pnpm installation and setup process
+func (exec *Execute) setupPnpm(pm *PackageManager) error {
+	// Check pnpm installation only once per Execute instance
+	if !exec.pnpmSetup.checked {
+		exec.pnpmSetup.checked = true
+		execRunner := exec.Utils.GetExecRunner()
+
+		// If a specific pnpm version is requested, always install and use it locally
+		if exec.Options.PnpmVersion != "" {
+			// Set up local pnpm path
+			if exec.pnpmSetup.command == "" {
+				absolutePnpmPath := filepath.Join(exec.pnpmSetup.rootDir, tmpInstallFolder, "node_modules", ".bin", "pnpm")
```

Comment:
Since these lines are repetitive , maybe its better to move it out to beginning of function .
We can always set absolutePnpmPath and exec.pnpmSetup.command = absolutePnpmPath
and only override when needed
Improves code readability with one less if else statement